docs/interfaces/mx4/leds.md

```diff
@@ -1 +1,187 @@
+
+index = 2 - GSM LED
+index = 3 - GPS LED
+index = 4 - FUNC LED
+index = 5 - BUZZER
+
+color = 0 - led off
+color = 1 - blink off/green
+color = 2 - blink off/orange
+color = 3 - blink green/orange
+
+freq = 0 - no blink
+freq = 1-255 - Flashing with frequency; freq * 100 milliseconds
+```
+### Pre configured LED behavior
+
+#### PWR
+
+The power led is is green or flashing green yellow/orange if charging an internal battery. Depending on the level of charge it will change speed between the flashing green<-->yellow, until it is lit green. If the unit does not have a battery it will continue flash green<-->yellow.
+
+- Solid green when running on supply voltage and battery charge state is above 60 %.
+- Flashing green/yellow 1 Hz: Battery charge status is 41-60 % and charging.
+- Flashing green/yellow 2 Hz: Battery charge status is 21-40 % and charging.
+- Flashing green/yellow 4 Hz: Battery charge status is 0-20 % and charging.
+- Solid yellow when running on battery and the battery is above 60 %
+- Flashing yellow/off 1 Hz: Battery charge status is 41-60 %.
+- Flashing yellow/off 2 Hz: Battery charge status is 21-40 %.
+- Flashing yellow/off 4 Hz: Battery charge status is 0-20 %.
+- Flashing green/off 1 Hz: The unit is in boot process
+
+#### STANDBY STATE
+- One-shot PWR LED green/off (on 30 ms) approximately every 20 seconds.
+
+#### ERROR STATE
+- All LEDS(not wifi) Flashing green/off 5 Hz: The unit is in an error state. Failing to boot operating system. This is the default state when the unit is powered on.
+
+#### FIRMWARE UPGRADE
+A firmware upgrade can be followed by monitoring the LED behavior.
+
+1. All LEDS(not wifi) Flashing green/off 1 Hz: The unit is loading/running an image ("hmupdate.img") from USB driver or /boot. DO NOT REMOVE POWER!
+1. When hmupdate.img is finished the system will start up. This will be indicated by PWR LED Flashing green/off 1 Hz.
+1. First boot after hmupdate.img is run we also update co-processor firmware. This is indicated by following sequence:
+- FUNC LED Solid green and other LED's off. Entered co-processor bootloader and will load an application to update co processor bootloader.
+- FUNC LED solid green, GPS solid green and other LED's off. Bootloader application is loaded and will update co-processor bootloader.
+- FUNC LED Solid green and other LED's off. Entered co-processor bootloader and will load the final application.
+- LED behavior returns to normal operation.
+
+During the above steps GSM LED will flash when we are sending data to the co-processor for upgrade.
+
+1. Done!
+
+## Set LEDs (the hard way)
+
+LED flashing is configured by sending 4 bytes on corresponding led_-file in Linux.
+
+**The configuration of the 4 bytes:**
+<table>
+    <tr>
+        <td>MSB</td>
+        <td>Second MSB</td>
+        <td>Second LSB</td>
+        <td>LSB</td>
+    </tr>
+    <tr>
+       <td>Frequency</td>
+       <td>Spare</td>
+       <td>Flash config</td>
+       <td>Current LED</td>
+   <tr>
+</table>
```

Comment:
If this doesn't render correctly you could try making a markdown table

Reply:
i will try to fix it, it is from mx4-wiki.